### DISCUSSION TITLE
Multiple IF, THEN, ELSE based on date

### DISCUSSION THREAD

**Discussion by au8st** [Budibase/budibase#12329] [Help] [answered]
Hi and apologies if this has been answered previously, however I'm not finding any answered via search.

I'm switching from NOCODB and I'm currently in the BUDIBASE data tab.  I want a formula field called `next service date` linked to a options field `service type` and based on the option the `next service date` will display when the next service is due:

- `service type` = Weekly / `next service date` = Next 7 days
- `service type` = Monthly / `next service date` = Next 28 days
- `service type` = 6 Monthly / `next service date` = Next 182 days
- `service type` = 12 Monthly / `next service date` = Next 365 days

I'll admit I'm brand new to handles and have touched on a little documentation but I learn more by example and hoping this is possible for what I'm aiming to achieve.

Thank you in advance.

**Comment by melohagan** [comment, marked as answer]
Hey @au8st 

When it comes to dates, I would recommend using JavaScript as it'll give you more options. Then it's a matter of looking up StackOverflow or even filtering these discussions by the `date` label to see some examples that could give you a starting point.

**Comment by au8st** [comment]
@melohagan, Thank you for the reply, I understand that to go further with no/low code platforms it appears JS is a fundamental requirement.  Looks like for the ideas I have and the complexity ahead it only seems logical to get to grasp with the language itself.

Many thanks for your time
